### PR TITLE
Refactor module syntax to use CodeRef to identify card being modified 

### PR DESCRIPTION
**Pull request by habdelra**
The issue here is that the ModuleSyntax did not operate on CodeRefs, rather it used a very simple mechanism to identify a declaration in the module that is a card that is being modified--either the exported name of the card (which lines up nicely for CodeRef's of exported cards), or the module scoped declaration of a card for non-exported cards. it didn't know how to resolve a CodeRef because you needed to have a running card in order to resolve a CodeRef.

The fix was to build a mechanism that works like `@cardstack/runtime-common/code-ref.ts#loadCard()` to identify a card, but to use syntax instead of a running card to actually operate.

adding-fields-bug

### CODE REVIEW

File: packages/runtime-common/module-syntax.ts [hunk 1]
```diff
@@ -193,30 +196,76 @@ export class ModuleSyntax {
...
+      );
+    } else {
+      if (classRef.classIndex == null) {
+        throw new Error(
+          `Cannot resolve class reference with undefined 'classIndex' when looking up interior card/field in module ${this.url.href}`,
+        );
+      }
+      let declaration = this.declarations[classRef.classIndex];
+      return this.possibleCardsOrFields.find(
+        (c) => c.path === declaration.path,
```
Comment (tintinthong) [inline packages/runtime-common/module-syntax.ts]: hmmmm. Is this code path being reached? 

In any case, classIndex is the index of the card of the possibleCardsOrFields array. Think we should just reference the possibleCardsOrFields array.

Comment (habdelra) [reply packages/runtime-common/module-syntax.ts]: It totally is—this code path is what is executed when the code ref is ancestorOf and fieldOf. You can see this in the module syntax unit tests, as well as that is literally what is being demoed in the screen capture

Comment (habdelra) [reply packages/runtime-common/module-syntax.ts]: I think you are getting the `declaration` confused with the `possibleCardOrField`. These are different things. If you follow the types, you’ll see that the syntax analysis refers to declarations for the super card or the field card, but what we actually want to return is the possibleCardOrField that corresponds to the declaration—that’s what getCard() is supposed to do.

Comment (habdelra) [reply packages/runtime-common/module-syntax.ts]: Or actually looking at this more closely I think you are right, I’m the one getting confused—I think im not trying this on a complex enough module. All the modules I’m trying this on only have declarations of cards. I’ll add another unit test that includes non card declarations too that appear before the cards.

**Review by tintinthong — APPROVED** [review]
👍

**Comment by habdelra** [comment]
`percySnashot()` looks like it is timing out in the tests which is causing some unrelated tests to fail.... ugh...

**Comment by habdelra** [comment]
for some reason percy dropped a huge pile of our snapshots. i added them back in....

**Pull request merged**